components/clp-py-utils/clp_py_utils/profiling_utils.py

```
@@ -0,0 +1,263 @@
+                    func, args, kwargs, job_id_param, task_id_param
+                )
+
+                profiler = Profiler(interval=PROFILING_INTERVAL)
+                try:
+                    profiler.start()
+                except RuntimeError as e:
+                    # Skip profiling this function to avoid conflicts
+                    if "already a profiler running" in str(e):
+                        logger.debug(
+                            f"Skipping nested profiling for {name} "
+                            f"(parent profiler already active)"
```

Ruff (0.14.0)

Logging statement uses f-string

(G004)